### ISSUE TITLE
Show NXentry title in tree

### ISSUE THREAD

**Issue by woutdenolf** [silx-kit/h5web#1622]
NXentry names are often adapted for machine consumption while the title is meant for human consumption.

Silx view shows both

![image](https://github.com/silx-kit/h5web/assets/7264703/5e89368f-0311-4780-a4b0-cf4596ada3ef)

H5web does not

![image](https://github.com/silx-kit/h5web/assets/7264703/d177b4d9-d2b6-436f-9b3b-e8a3e86d934e)

It does mean you would need to fetch the "title" dataset if any which you might not want to do? Or perhaps make it optional?

**Comment by loichuder** [comment]
Hm. How far would you like to go with this ?

Because if I look at our beloved `water_224.h5` file in `silx view`, the description column can be filled in many ways:
![image](https://github.com/silx-kit/h5web/assets/42204205/a56c1a3d-3415-4f4b-abe5-eba541828780)

- NXentry `entry_0000`: `title` dataset
- NXprocess `1_integration`: `program` dataset
- NXinstrument `BM20`: `name` dataset
- NXsample `water`: `description` dataset

**Comment by axelboc** [comment]
> It does mean you would need to fetch the "title" dataset if any which you might not want to do? Or perhaps make it optional?

Indeed, for each "NX" badge, we have access only to the metadata of the group itself (i.e. `dataset01`, `dataset02`, etc.), including attributes, but not the metadata of its children. We could fetch the metadata of the children as well, but this may lead to requests pile-ups in files/groups with lots of NX sub-groups, thus slowing down fetching actual data. It gets worse when considering nested NX entities, as Loïc mentioned.

I could also imagine cases where the groups would be named meaningfuly and the titles would be generic...

Maybe there's a compromise to be found in naming the datasets, like `01_APS-13-ID-C_Co_K`? In a way, it's not really helping machines either (including H5Web) if, to get to a meaningful name, they have to know about NeXus and dig through the tree to find a distinctive piece of information. :stuck_out_tongue_closed_eyes:

**Comment by woutdenolf** [comment]
> How far would you like to go with this ?

Just the NXentry title.
